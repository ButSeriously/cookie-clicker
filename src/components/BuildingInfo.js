import React from 'react';
import PropTypes from 'prop-types';
import smallCookie from '../img/perfectCookie-small.png';

const BuildingInfo = props => (
  <div className="building_info">
    <div className="building_info__header">
      <div className="building_info__name">
        <p>{props.producentInfo.name}</p>
        <p>[owned: {props.producentInfo.amount}]</p>
      </div>
      <p className="building_info__cost">
        <img src={`./public/${smallCookie}`} alt="Cookie.png" />
        {props.producentInfo.cost}
      </p>
    </div>
    <p className="building_info__description">{props.producentInfo.description}</p>
    <ul className="building_info__stats">
      <li>• each farm produces {props.producentInfo.productionPerSecond} cookies per second</li>
      <li>• {props.producentInfo.amount} cursors producing&nbsp;
        {(props.producentInfo.amount * props.producentInfo.productionPerSecond).toFixed(1)}&nbsp;
        cookies per second
      </li>
    </ul>
  </div>
);

BuildingInfo.propTypes = {
  producentInfo: PropTypes.object.isRequired,
};

export default BuildingInfo;
