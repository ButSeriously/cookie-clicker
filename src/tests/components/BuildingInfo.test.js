import React from 'react';
import { shallow } from 'enzyme';
import BuildingInfo from '../../components/BuildingInfo';
import stats from '../fixtures/buildingStats';

test('should render BuildingInfo', () => {
  const wrapper = shallow(<BuildingInfo buildingInfo={stats} />);
  expect(wrapper).toMatchSnapshot();
});
