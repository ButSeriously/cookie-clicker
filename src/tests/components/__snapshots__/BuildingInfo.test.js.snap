// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BuildingInfo 1`] = `
<div
  className="building_info"
>
  <div
    className="building_info__header"
  >
    <div
      className="building_info__name"
    >
      <p>
        Cursor
      </p>
      <p>
        [owned: 
        12
        ]
      </p>
    </div>
    <p
      className="building_info__cost"
    >
      <img
        alt="Cookie.png"
        src="./public/test-file-stub"
      />
      15
    </p>
  </div>
  <p
    className="building_info__description"
  >
    Autoclicks once every 10 seconds.
  </p>
  <ul
    className="building_info__stats"
  >
    <li>
      • each farm produces 
      0.1
       cookies per second
    </li>
    <li>
      • 
      12
       cursors producing 
      1.2
        cookies per second
    </li>
  </ul>
</div>
`;
